app/electron/main.ts

```diff
@@ -1087,6 +1156,72 @@ async function getRunningHeadlampPIDs() {
   return processes.map(pInfo => pInfo.pid);
 }
 
+/**
+ * Check if a specific port is occupied by a Headlamp process
+ * @returns Array of Headlamp PIDs using the port, or null if port is free or used by another process
+ */
+async function getHeadlampPIDsOnPort(port: number): Promise<number[] | null> {
+  try {
+    // First check if the port is in use at all
+    const portInUse = !(await isPortAvailable(port));
+    if (!portInUse) {
+      console.debug(`Port ${port} is available (not in use)`);
+      return null; // Port is free
+    }
+
+    console.debug(`Port ${port} is in use, checking if it's Headlamp...`);
+
+    // Get all Headlamp processes
+    const headlampProcesses = await find_process('name', 'headlamp-server');
+    console.debug(
+      `Found ${headlampProcesses.length} Headlamp process(es):`,
+      headlampProcesses.map(p => ({ pid: p.pid, name: p.name, cmd: p.cmd }))
+    );
+
+    if (headlampProcesses.length === 0) {
+      console.debug(`No Headlamp processes running`);
+      return null; // No Headlamp processes running
+    }
+
+    // Parse command line arguments to find which Headlamp process is using this port
+    const headlampOnPort = headlampProcesses.filter(p => {
+      if (!p.cmd) return false;
+
+      // Look for --port=XXXX or --port XXXX in the command line
+      const portRegex = /--port[=\s]+(\d+)/;
+      const match = p.cmd.match(portRegex);
+
+      if (match && match[1]) {
+        const processPort = parseInt(match[1], 10);
+        console.debug(`Process ${p.pid} is using port ${processPort}`);
+        return processPort === port;
+      }
+
+      // If no port specified, Headlamp uses default port 4466
+      if (port === 4466 && !p.cmd.includes('--port')) {
+        console.debug(`Process ${p.pid} is using default port 4466`);
+        return true;
+      }
+
+      return false;
+    });
+
+    if (headlampOnPort.length === 0) {
+      console.debug(`Port ${port} is used by non-Headlamp process`);
+      return null; // Port is used by non-Headlamp process
+    }
+
+    console.info(
+      `Port ${port} is occupied by Headlamp process(es):`,
+      headlampOnPort.map(p => p.pid)
+    );
+    return headlampOnPort.map(p => p.pid);
+  } catch (error) {
+    console.error(`Error checking if port ${port} is used by Headlamp:`, error);
```

Comment:
Remove "Headlamp"? Because it might be called something else.

(and in other parts of this PR/file too)